src/lang/std/engineConnection.ts

```diff
@@ -1936,7 +1936,14 @@ export class EngineCommandManager extends EventTarget {
       command,
       idToRangeMap: {},
       range: [0, 0],
-    }).then(([a]) => a)
+    })
+      .then(([a]) => a)
+      .catch((e) => {
+        // TODO: Previously was never caught, since adding rejectAllCommands how should we handle this
+        // error?
+        /*noop*/
+        return null
+      })
```

Comment:
I think swallowing rejected scene commands is probably fine.

maybe 
```ts
if (e.msg ===
        'Failed to wait for promise from engine: JsValue("Force interrupt, executionIsStale, new AST requested")') {
  // swallow error
} else {
  // console.errror?
}
```

Reply:
No this won't work sorry.
send scene commands have nothing to do with rust, that's why you rejecting all pending commands causes this error. Mabye we need to differentiate between scene and modeling commands in pending commands and then do `rejectAllModelingCommands` instead of all modeling commands.

Reply:
I think you are correct on this. I was able to break the `initPlanes` commands in some workflows when spamming file switching or if the connection to the engine refreshed.

Can I use that type to see if they fall into the `Models['ModelingCmd_type]` ?

Reply:
hmm so to clarify the modeling commands and scene commands is only a differentiation that we using in the modeling app, from the engine's perspective everything is a modeling command (with the exception of like webrtc setup command etc). Point is you're not going to be able to use types.

The break down is basically modeling commands are commands made by the rust execution
and everything the modeling app sends directly (mostly camera stuff) is a scene command. I'm being lazy in just sending diff screenshots, but this is what I think we should do

![image](https://github.com/user-attachments/assets/7bd0c780-b2f9-4fe2-80d5-f4ee024e243d)

The type will need this added too

![image](https://github.com/user-attachments/assets/856dc328-4d27-4af7-b790-402f1047169e)

Than update your rejectAllCommands to something like
```ts
rejectAllModelingCommands(rejectionMessage: string) {
    Object.values(this.pendingCommands).forEach(
      ({ reject, isSceneCommand }) => !isSceneCommand && reject(rejectionMessage)
    )
  }
```

Does that make sense?

Reply:
Yup this makes sense, thanks for the detailed outline.